File: php/features.php
<?php
declare(strict_types=1);

// MetMuseumCollection SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MetMuseumCollectionFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MetMuseumCollectionBaseFeature();
            case "test":
                return new MetMuseumCollectionTestFeature();
            default:
                return new MetMuseumCollectionBaseFeature();
        }
    }
}

File: py/features.py
# MetMuseumCollection SDK feature factory

from feature.base_feature import MetMuseumCollectionBaseFeature
from feature.test_feature import MetMuseumCollectionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MetMuseumCollectionBaseFeature(),
        "test": lambda: MetMuseumCollectionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
